core/src/main/java/hudson/model/Queue.java

```diff
@@ -2498,33 +2498,25 @@ protected NotWaitingItem(NotWaitingItem ni) {
     public final class BlockedItem extends NotWaitingItem {
         private transient CauseOfBlockage causeOfBlockage = null;
 
-        /**
-         * @deprecated Use {@link #BlockedItem(WaitingItem, CauseOfBlockage)} instead
```

Comment:
Or just leave it deprecated and say that outside code is not supposed to be creating this stuff to begin with.